core/src/main/java/org/elasticsearch/env/Environment.java

```diff
@@ -108,28 +109,32 @@
     }
 
     public Environment(Settings settings) {
+        this(PathUtils.getDefaultFileSystem(), settings);
+    }
+
+    public Environment(FileSystem fs, Settings settings) {
```

Comment:
> it will for example cause tests that are running on a filesystem that handles paths differently to fail 

But we shouldn't be testing every FS. We shoudl be testing the logic in plugin installation, which is either is or is not posix. So you can use posix for testing the stuff works, and "basic" for testing it skips the extra code gracefully for non-posix FS.